internal/venice-consumer/src/main/java/com/linkedin/venice/client/consumer/VeniceChangeLogConsumerImpl.java

```diff
@@ -0,0 +1,288 @@
+      }
+    }
+    return pubSubMessages;
+  }
+
+  private PubSubMessage<K, V, Long> convertRecordToPubSubMessage(
+      KafkaMessageEnvelope kafkaMessageEnvelope,
+      K currentKey,
+      PubSubTopicPartition pubSubTopicPartition,
+      Long offset,
+      Long timestamp,
+      int payloadSize) {
+    MessageType messageType = MessageType.valueOf(kafkaMessageEnvelope.messageType);
+    V currentValue = null;
+    switch (messageType) {
+      case PUT:
+        Put put = (Put) kafkaMessageEnvelope.payloadUnion;
+        Schema valueSchema = schemaReader.getValueSchema(put.schemaId);
+        RecordDeserializer<V> valueDeserializer =
+            FastSerializerDeserializerFactory.getFastAvroGenericDeserializer(valueSchema, valueSchema);
+        currentValue = valueDeserializer.deserialize(put.putValue);
+        break;
+    }
+    return new ImmutablePubSubMessage(currentKey, currentValue, pubSubTopicPartition, offset, timestamp, payloadSize);
+  }
+
+  private PubSubMessage<K, ChangeEvent<V>, Long> convertChangeEventToPubSubMessage(
+      Put put,
+      K currentKey,
+      PubSubTopicPartition pubSubTopicPartition,
+      Long offset,
+      Long timestamp,
+      int payloadSize) {
+    byte[] valueBytes = put.putValue.array();
+    RecordChangeEvent recordChangeEvent = recordChangeDeserializer.deserialize(valueBytes);
+    Schema currentValueSchema = schemaReader.getValueSchema(1);
+    RecordDeserializer<V> valueDeserializer =
+        FastSerializerDeserializerFactory.getFastAvroGenericDeserializer(currentValueSchema, currentValueSchema);
+    V currentValue = null;
+    if (recordChangeEvent.currentValue != null) {
+      currentValue = valueDeserializer.deserialize(recordChangeEvent.currentValue.getValue());
+    }
+    V previousValue = null;
+    if (recordChangeEvent.previousValue != null) {
+      previousValue = valueDeserializer.deserialize(recordChangeEvent.previousValue.getValue());
+    }
+    ChangeEvent<V> changeEvent = new ChangeEvent<>(previousValue, currentValue);
+    return new ImmutablePubSubMessage(currentKey, changeEvent, pubSubTopicPartition, offset, timestamp, payloadSize);
+  }
+
+  private void handleControlMessage(ControlMessage controlMessage) {
+    final ControlMessageType controlMessageType = ControlMessageType.valueOf(controlMessage);
+    if (controlMessageType.equals(ControlMessageType.VERSION_SWAP)) {
+      VersionSwap versionSwap = (VersionSwap) controlMessage.controlMessageUnion;
+      LOGGER.info("Obtain version swap message: " + versionSwap);
+      String newServingVersionTopic = versionSwap.newServingVersionTopic.toString();
+      Set<Integer> partitions = new HashSet<>();
+      for (Integer partition: subscribedPartitions) {
+        partitions.add(partition);
+      }
+      unsubscribe(subscribedPartitions);
+      if (viewClassName.equals(ChangeCaptureView.class.getCanonicalName())) {
+        currentTopic = newServingVersionTopic + ChangeCaptureView.CHANGE_CAPTURE_TOPIC_SUFFIX;
```

Comment:
oof.  This seems like an oversight (on my part).  It probably would have made more sense to have the topicSwitch message contain the name of the topic that we want to traverse to.  Oh well.